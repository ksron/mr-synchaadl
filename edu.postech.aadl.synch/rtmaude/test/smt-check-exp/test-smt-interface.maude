set show timing off .

load ../../interpreter-smt-check-symbolic.maude

select SMT-CHECKSAT .
red check-sat(r(0) < [3/1] and r(0) > [1/1]) .
red check-sat(r(0) < [3/1] and r(0) > [1/1]) .

red check-sat(r("a.b",0) < [3/1] and r("a.b",0) > [1/1]) .
red check-sat(r("a",0) < [3/1] and r("b",0) > [1/1]) .

red check-sat(r(0) < [3/1] and r(0) > [1/2]) .
red check-sat(r(0) < [3/7] and r(0) > [1/2]) .

red check-sat(r("a.b",0) < [3/1] and r("a.b",0) > [1/2]) .
red check-sat(r("a",0) < [3/7] and r("b",0) > [1/2]) .
red check-sat(r("a",0) < [3/7] and r("a",0) > [1/2]) .
