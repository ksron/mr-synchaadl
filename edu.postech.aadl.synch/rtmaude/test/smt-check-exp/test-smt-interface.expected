==========================================
reduce in SMT-CHECKSAT : check-sat(r(0) < [(3).NzNat] and r(0) > [(1).NzNat]) .
rewrites: 38
result Bool: (true).Bool
==========================================
reduce in SMT-CHECKSAT : check-sat(r(0) < [(3).NzNat] and r(0) > [(1).NzNat]) .
rewrites: 30
result Bool: (true).Bool
==========================================
reduce in SMT-CHECKSAT : check-sat(r("a.b", 0) < [(3).NzNat] and r("a.b", 0) >
    [(1).NzNat]) .
rewrites: 34
result Bool: (true).Bool
==========================================
reduce in SMT-CHECKSAT : check-sat(r("a", 0) < [(3/1).Rat] and r("b", 0) > [(
    1).NzNat]) .
rewrites: 35
result Bool: (true).Bool
==========================================
reduce in SMT-CHECKSAT : check-sat(r(0) < [(3).NzNat] and r(0) > [(1/2).Rat]) .
rewrites: 38
result Bool: (true).Bool
==========================================
reduce in SMT-CHECKSAT : check-sat(r(0) < [(3/7).Rat] and r(0) > [(1/2).Rat]) .
rewrites: 38
result Bool: (false).Bool
==========================================
reduce in SMT-CHECKSAT : check-sat(r("a.b", 0) < [(3).NzNat] and r("a.b", 0) >
    [(1/2).Rat]) .
rewrites: 34
result Bool: (true).Bool
==========================================
reduce in SMT-CHECKSAT : check-sat(r("a", 0) < [(3/7).Rat] and r("b", 0) > [(
    1/2).Rat]) .
rewrites: 34
result Bool: (true).Bool
==========================================
reduce in SMT-CHECKSAT : check-sat(r("a", 0) < [(3/7).Rat] and r("a", 0) > [(
    1/2).Rat]) .
rewrites: 34
result Bool: (false).Bool
Bye.
